src/python/WMCore/Services/Rucio/Rucio.py

```diff
@@ -356,29 +357,40 @@ def attachDIDs(self, rse, superDID, dids, scope='cms'):
              then it's a container name; if attaching files, then it's a block name)
         :param dids: either a string or a list of data identifiers (can be block or files)
         :param scope: string with the scope name
+        :param chunkSize: number of dids to send to upstream Rucio server
         :return: a boolean to represent whether it succeeded or not
         """
         if not isinstance(dids, list):
             dids = [dids]
         # NOTE: the attaching dids do not create new container within a scope
         # and it is safe to use cms scope for it
-        dids = [{'scope': 'cms', 'name': did} for did in dids]
+        alldids = [{'scope': 'cms', 'name': did} for did in sorted(dids)]
 
-        response = False
-        try:
-            response = self.cli.attach_dids(scope, superDID, dids=dids, rse=rse)
-        except DuplicateContent:
-            self.logger.warning("Dids: %s already attached to: %s", dids, superDID)
-            response = True
-        except FileAlreadyExists:
-            self.logger.warning("Failed to attach files already existent on block: %s", superDID)
-            response = True
-        except DataIdentifierNotFound:
-            self.logger.error("Failed to attach dids: %s. Parent DID %s does not exist.", dids, superDID)
-        except Exception as ex:
-            self.logger.error("Exception attaching dids: %s to: %s. Error: %s",
-                              dids, superDID, str(ex))
-        return response
+        for dids in getChunk(alldids, chunkSize):
```

Comment:
Before this for loop, it might be useful to have a code like:
```
if len(alldids) > chunkSize:
    self.logger.info("Attaching a total of %d DIDs in chunk size of: %d", len(alldids), chunkSize)
```

Reply:
ok, done